src/compositions/usePagination.ts

```diff
@@ -0,0 +1,202 @@
+import { Getter } from '@prefecthq/prefect-design'
+import { SubscriptionOptions, useSubscription, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+import { ComputedRef, MaybeRef, Ref, computed, ref, watch } from 'vue'
+import { GLOBAL_API_LIMIT } from '@/compositions/useFilterPagination'
+import { UseSubscriptions, useSubscriptions } from '@/compositions/useSubscriptions'
+import { repeat } from '@/utilities/arrays'
+
+type PaginationFilter = {
+  limit?: number,
+  offset?: number,
+}
+
+// typescript only lets you use any here
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PaginationParameters = [filter?: PaginationFilter, ...any[]]
+type PaginationFetchAction = (...parameters: PaginationParameters) => Promise<unknown[]>
+type PaginationCountAction = (...parameters: PaginationParameters) => Promise<number>
+
+type FetchSubscriptionAction<
+  TFetch extends PaginationFetchAction
+> = (parameters: Parameters<TFetch>[]) => Promise<Awaited<ReturnType<TFetch>>>
+
+export type PaginationOptions = SubscriptionOptions & {
+  mode?: 'page' | 'infinite',
+  page?: MaybeRef<number>,
+}
+
+export type UsePaginationParameters<
+  TFetch extends PaginationFetchAction,
+  TFetchParameters extends Getter<Parameters<TFetch> | null>,
```

Comment:
I went that route originally because it was just simpler. But that might have been an issue with a previous iteration. 

I feel like using this composition with a static filter isn't super useful. But we could allow for it if needed.